Sources/ForTag.swift

```diff
@@ -124,25 +131,115 @@ class ForNode : NodeType {
 
     if !values.isEmpty {
       let count = values.count
+      var result = ""
+
+      var currentLoopContext = context["forloop"] as? [String: Any]
 
-      return try values.enumerated().map { index, item in
-        let forContext: [String: Any] = [
+      for (index, item) in values.enumerated() {
+        var forContext: [String: Any] = [
           "first": index == 0,
           "last": index == (count - 1),
           "counter": index + 1,
-          "counter0": index,
-        ]
+          "counter0": index
+          ]
+        if let label = label {
+          forContext[label] = forContext
+          forContext["label"] = label
+        }
+        if let label = currentLoopContext?["label"] as? String,
```

Comment:
If I understand this correctly, this will only work for 1 parent labeled loop. If I have more than 1 parent labeled loop, I won't be able to access their context.

I'm not saying we should support such a use case (or not), but if we don't, we should document it clearly.